src/utils/streams.js

```diff
@@ -0,0 +1,118 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+// This file holds various utilities about streams.
+
+import { Transform, Writable } from 'stream';
+import crypto from 'crypto';
+
+import { getLogger, type Logger } from '../log';
+
+/**
+ * This transform computes a sha1 of the daa passing through it, but otherwise
+ * doesn't alter the data.
+ */
+export class HasherPassThrough extends Transform {
+  log: Logger = getLogger('HasherPassThrough');
+  sha1Hasher: crypto$Hash = crypto.createHash('sha1');
+  sha1Result: string | null = null;
+
+  _transform(chunk: *, encoding: *, callback: *) {
```

Comment:
I checked that using `*` made the typing work properly. I didn't type them explicitely because it's a bit cumbersome :( We inherit them from the parent class and Flow should be able to find them automatically.
Actually it wasn't requiring me to add them (not even the star type) before I've put them to a separate file. And if we put the wrong types it also errors. I find Flow is super dumb here.